File: java/byps/src-ureq/byps/ureq/BUtilityRequests.java
package byps.ureq;

import java.util.Map;

import byps.BRemote;
import byps.RemoteException;

/**
 * Interface with internally used utility functions.
 * 
 * Hint: This API does not support versioning.   
 */
public interface BUtilityRequests extends BRemote {
  
  public void cancelMessage(long messageId) throws RemoteException;
  
  public Map<String,String> testAdapter(String functionName, Map<String,String> params) throws RemoteException;
  
  public Map<String,String> execute(String functionName, Map<String,String> params) throws RemoteException;
  
}

File: csharp/byps/ureq/byps/ureq/BUtilityRequests.cs
using System;
using System.Collections.Generic;
using byps;

namespace byps.ureq
{
	
	/// <summary>
	/// Interface with internally used utility functions.
	/// </summary>
	/// <remarks>
	/// Hint: This API does not support versioning.
	/// </remarks>
	// checkpoint byps.gen.cs.GenRemoteClass:93
	public interface BUtilityRequests : BRemote {
	
		void CancelMessage(long messageId);
		void CancelMessage(long messageId, BAsyncResult<Object> asyncResult) ;
		
		IDictionary<String,String> TestAdapter(String functionName, IDictionary<String,String> @params);
		void TestAdapter(String functionName, IDictionary<String,String> @params, BAsyncResult<IDictionary<String,String>> asyncResult) ;
		
		IDictionary<String,String> Execute(String functionName, IDictionary<String,String> @params);
		void Execute(String functionName, IDictionary<String,String> @params, BAsyncResult<IDictionary<String,String>> asyncResult) ;
		
		
	}
}  // end namespace
